Add prosemirror dependacies, types for client/ui/prosemirror.mjs, and foundry/scripts/prosemirror.js

Spice-King opened the pull request:
This should be everything for prosemirror, ignoring #2088 as the one external bit that touches it, closes #2092.

kmoschcau commented:
Some of the methods in those classes should be marked as `protected` and some in addition also marked as `@internal` in their doc. This mostly affects methods starting with an underscore. There's a pinned post by me in the discord channel, that should still be up to date.

Spice-King commented:
Rebased, added `protected`, but waiting on clarification about `@internal` use noted [here](https://discord.com/channels/732325252788387980/793933527065690184/1004929455573180497).

Spice-King commented:
Removed unneeded `@internal` notes, updated to 10.277

kmoschcau commented on src/foundry/client/head.d.ts:
```diff
@@ -1,3 +1,4 @@
+import * as ProseMirrorObject from '../prosemirror/prosemirror';
 export {};
```
This can now be removed, that we have an import here. It was just to force this to be a module.

kmoschcau commented on src/foundry/prosemirror/menu.d.ts:
```diff
@@ -0,0 +1,206 @@
+   */
+  protected _insertLinkPrompt(): Promise<void>;
+
+  /* -------------------------------------------- */
```
Remove this please.

kmoschcau commented on src/foundry/prosemirror/util.d.ts:
```diff
@@ -0,0 +1,31 @@
+import type { Node, Schema } from 'prosemirror-model';
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { StringNode } from './string-serializer';
```
What is this for?

Spice-King replied:
Brought in strictly to make VS Code happy rendering `{@link StringNode#toString}` right as a link, as it's not a global class instance. Do you know another way to do that?

kmoschcau replied:
Oh weird I didn't know VSC needed that. In that case we have a nice tooling conflict here. It's fine to leave it in, but can you please add a comment why this is done? Then it's good from my side.

Spice-King replied:
Comment added.
